features/learn/src/main/kotlin/com/teacherworkout/features/learn/data/Result.kt

```diff
@@ -0,0 +1,6 @@
+package com.teacherworkout.features.learn.data
+
+sealed class Result<out R> {
+    data class Success<out T>(val data: T) : Result<T>()
+    data class Error(val exception: Exception) : Result<Nothing>()
```

Comment:
Firstly, instead of `R`, it should have been `T`. My bad. `T` would be the object that we expect from the API call. If the API call throws an exception, I don't think that we would still get a `T` object as a response. Anyway, this `Result` class might not be complete yet.